### ISSUE TITLE
Support for test.fail

### ISSUE THREAD

**Issue by alirezamirian** [cypress-io/cypress#23687]
### What would you like?

 A way to specify a test must fail. That, compared to `.skip` would be similar to how [@ts-expect-error vs @ts-ignore](https://typescript-eslint.io/rules/prefer-ts-expect-error/).

It can be similar to Playwright, which supports a [test.fail()](https://playwright.dev/docs/api/class-test#test-fail-1) function to be called in test. It can also be similar to `test.skip`, I guess.

### Why is this needed?

If `.skip` is used to exclude a test that is meant to capture a bug that is supposed to be fixed at some point, it's easy to forget about removing `.skip` annotation when the bug is fixed.

### Other

Not sure if it should be a feature request here or on Mocha.

**Comment by nagash77** [comment]
@alirezamirian Can you provide an example test of a use case where you think this feature would solve?

**Comment by alirezamirian** [comment]
@nagash77 Sure. 
Suppose you identify a bug, create a test case for it, but can't be fixed now for some reason (maybe requires a fix in a dependency, you are waiting for). Now you can just `.skip` it and when the fix is in place, you can simply remove `.skip`. But it's likely to forget about the skipped test, and not unskip it when the fix is applied. By using `.fail` you make sure you won't forget about it, when the issue is fixed. Very similar to how [@ts-expect-error is prefered over @ts-ignored](https://typescript-eslint.io/rules/prefer-ts-expect-error/). As suggested in the [Playwright documentation](https://playwright.dev/docs/api/class-test#test-fail-1:~:text=This%20is%20useful%20for%20documentation%20purposes%20to%20acknowledge%20that%20some%20functionality%20is%20broken%20until%20it%20is%20fixed.) as well, "This is useful for documentation purposes to acknowledge that some functionality is broken until it is fixed."

**Comment by lmiller1990** [comment]
You could just monkey patch it (which is what we'd be doing anyway, if we implemented it).

Untested, but this is the idea:

```js
it.fail = function (msg) {
  it(msg, () => { 
    throw Error('Not implemented')
  })
}
```

You could put it in your `supportFile` to make it available in every test.

**Comment by alirezamirian** [comment]
> ```js
> it.fail = function (msg) {
>   it(msg, () => { 
>     throw Error('Not implemented')
>   })
> }
> ```

I don't think that would be a solution. Note that the purpose of `.fail` is to run the test, and ensure it fails. I haven't thought much about a user-land implementation, but even if it's possible, it should involve **running the actual test**, and somehow inverting the test success status, maybe by a wrapping `expect`. Or maybe it would require a lower-level API, not exposed by the runner.

**Comment by lmiller1990** [comment]
Right, I understand the feature request now.

I think we can still do this in userland/plugin for now, that way you can have the feature soon. Even if we do end up implementing this in core, it'll take some time/effort, and some initial exploration in the form of a user land solution would be a good place to start prototyping to understand the full surface area and complexity of the feature.

**Comment by lmiller1990** [comment]
Okay, it doesn't seem possible to "properly" work it in from the outside - this would need to implemented in **mocha** (I don't think it's even practical to implement it in Cypress, without completely hacking into Mocha... not ideal).

Here's my best effort:

```ts
// @ts-ignore
it.fail = async function (msg: string, cb: () => any) {
  let failed = false

  try {
    it.skip(msg, () => {})
    await cb()
  } catch (e) {
    failed = (e as Error).name === 'AssertionError' 
  }

  if (!failed) {
    new Function(
      `it("${msg}", () => {
        throw new this.AssertionError('Expected test to fail, but it passed')
      })`
    ).call({ AssertionError })
  }
}
```

It skips the initial test, and dynamically makes a new one to fail, which is kind of what you want (except you end up with one extra skipped test, but this doesn't really cause and problems):

<img width="454" alt="image" src="https://user-images.githubusercontent.com/19196536/189043501-13fbdbab-8cdc-4bb7-8dcd-dff9e7d35025.png">

Happy to leave this open, but if we did decide to implement it, we'd probably need to contribute to Mocha, which means getting their maintainers on board.

**Comment by romankhomitskyi** [comment]
Just add .skip and TODO comment and link jira issue
📈
